controllers/machine_controller.go

```diff
@@ -646,39 +646,57 @@ func (r *MachineReconciler) nodeToMachine(o client.Object) []reconcile.Request {
 		panic(fmt.Sprintf("Expected a Node but got a %T", o))
 	}
 
-	// Match by nodeName and status.nodeRef.name.
-	filters := []client.ListOption{
-		client.MatchingFields{clusterv1.MachineNodeNameIndex: node.Name},
-	}
-
+	var filters []client.ListOption
 	// Match by clusterName when the node has the annotation.
 	if clusterName, ok := node.GetAnnotations()[clusterv1.ClusterNameAnnotation]; ok {
-		filters = append(filters,
-			client.MatchingLabels{
-				clusterv1.ClusterLabelName: clusterName,
-			},
-		)
+		clusterNameFilter := client.MatchingLabels{
+			clusterv1.ClusterLabelName: clusterName,
+		}
+		filters = append(filters, clusterNameFilter)
```

Comment:
```suggestion
		filters = append(filters, client.MatchingLabels{
			clusterv1.ClusterLabelName: clusterName,
		})
```
(in case you want to change it, same for `namespaceFilter` below)
We don't need the local vars.